### PR TITLE
Devise token auth

### PR DESCRIPTION
**Pull request by MaaooVaass**
## summary
Implement authentication with devise-token-auth.

## Cards 
https://trello.com/c/RU3mYZwO/24-implementar-autenticaci%C3%B3n

### CODE REVIEW

File: app/models/model.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Model < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+end
```
Comment (fedeest) [inline app/models/model.rb]: Erase this, it's an example

File: db/migrate/20181101172146_devise_create_users.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      ## Database authenticatable
+      t.string :email,              null: false, default: ""
+      t.string :encrypted_password, null: false, default: ""
+
+      ## Recoverable
+      t.string   :reset_password_token
+      t.datetime :reset_password_sent_at
+
+      ## Rememberable
+      t.datetime :remember_created_at
```
Comment (fedeest) [inline db/migrate/20181101172146_devise_create_users.rb]: Your user is neither recoverable nor rememberable. This fields should not be included

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+            # Include default devise modules.
+            devise :database_authenticatable, :registerable
+                    #:recoverable, :rememberable, :trackable, :validatable,
```
Comment (fedeest) [inline app/models/user.rb]: I recommend adding trackable as well on real use cases. You can keep it this way if you like.

File: db/migrate/20181102224936_devise_token_auth_create_users.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    add_column :users, :current_sign_in_at, :datetime
+    add_column :users, :last_sign_in_at, :datetime
+    add_column :users, :current_sign_in_ip, :string
+    add_column :users, :last_sign_in_ip, :string
+    add_column :users, :confirmation_token, :string
+    add_column :users, :confirmed_at, :datetime
+    add_column :users, :confirmation_sent_at, :datetime
+    add_column :users, :unconfirmed_email, :string
+    add_index :users, [:uid, :provider], unique: true
+  end
```
Comment (fedeest) [inline db/migrate/20181102224936_devise_token_auth_create_users.rb:12]: Why are you adding all these columns here?

**Review by fedeest — CHANGES_REQUESTED** [review]

**Review by lmasello — APPROVED** [review]
Correct fede's corrections and it's good to go!

**Review by fedeest — APPROVED** [review]

**Pull request merged**